### PR TITLE
Add migration guide for v5->v6 transition

### LINKED ISSUE #2546: Create migration guide for AWS 6.0
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details
Create migration guide for AWS 6.0
Theoretically, the upgrade should be the same as upgrading to the upstream terraform provider version 5.x - we'll need to include all of the breaking changes from upstream [here](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.0.0)

### PR DESCRIPTION
**Pull request by iwahbe**
Fixes #2546

### CODE REVIEW

File: migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+- [v5.3.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.3.0)
+- [v5.4.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.4.0)
+- [v5.5.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.5.0)
+- [v5.6.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.6.0)
+- [v5.6.1](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.6.1)
+- [v5.6.2](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.6.2)
+- [v5.7.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.7.0)
+- [v5.8.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.8.0)
+
+### Removing depreciated resources
```
Comment (mnlumi) [inline migration-guide.md:18]: Are these resources fully removed or have they been aliased? From a user's perspective, what changes do they need to make?

Comment (iwahbe) [reply migration-guide.md:18]: They have been removed, in that they are no longer generated as resources and functions in downstream SDKs. They are type aliased in the schema, so users will not need to perform state surgery to continue when they move to the new definitions.

**Comment by iwahbe** [comment]
We should also link https://registry.terraform.io/providers/hashicorp/aws/latest/docs/guides/version-5-upgrade directly.

**Comment by mnlumi** [comment]
Moved to https://github.com/pulumi/registry/pull/2941

**Pull request closed without merging**